src/lib/lightnet.js

```diff
@@ -0,0 +1,703 @@
+  } catch (_) {
+    // Ignore the error if container doesn't exist
+  }
+}
+
+async function removeDockerContainer(containerName) {
+  try {
+    await shellExec(`docker rm ${containerName}`, { silent: true });
+  } catch (_) {
+    // Ignore the error if container doesn't exist
+  }
+}
+
+async function removeDockerVolume(volume) {
+  try {
+    await shellExec(`docker volume rm ${volume}`, { silent: true });
+  } catch (_) {
+    // Ignore the error if volume doesn't exist
+  }
+}
+
+async function removeDanglingDockerImages() {
+  await shellExec('docker image prune -f --filter "dangling=true"', {
+    silent: true,
+  });
+}
+
+async function saveDockerContainerProcessesLogs() {
+  const timeZoneOffset = new Date().getTimezoneOffset() * 60000;
+  const localMoment = new Date(Date.now() - timeZoneOffset);
+  const logsDir = `${lightnetLogsDir}/${
+    localMoment.toISOString().split('.')[0]
+  }`;
+  try {
+    const lightnetConfig = fs.readJSONSync(lightnetConfigFile);
+    const mode = lightnetConfig.mode;
+    const archive = lightnetConfig.archive;
+    await fs.ensureDir(logsDir);
+    if (mode === 'single-node') {
+      const logFilePaths = [
+        'logs/archive-node-api.log',
+        'logs/archive-node.log',
+        'logs/single-node-network.log',
+      ];
+      for (const logFilePath of logFilePaths) {
+        try {
+          await shellExec(
+            `docker cp ${lightnetDockerContainerName}:/root/${logFilePath} ${logsDir}/${logFilePath.replace(
+              /\//g,
+              '_'
+            )}`,
+            { silent: true }
+          );
+        } catch (_) {
+          // Ignore the error if log file doesn't exist
+        }
+      }
+    } else {
+      const logFilePaths = [
+        'archive/log.txt',
+        'fish_0/log.txt',
+        'node_0/log.txt',
+        'seed/log.txt',
+        'snark_coordinator/log.txt',
+        'snark_workers/worker_0/log.txt',
+        'whale_0/log.txt',
+        'whale_1/log.txt',
+      ];
+      if (archive) {
+        await shellExec(
+          `docker cp ${lightnetDockerContainerName}:/root/logs/archive-node-api.log ${logsDir}`,
+          { silent: true }
+        );
+      }
+      for (const logFilePath of logFilePaths) {
+        try {
+          await shellExec(
+            `docker cp ${lightnetDockerContainerName}:/root/.mina-network/mina-local-network-2-1-1/nodes/${logFilePath} ${logsDir}/${logFilePath.replace(
+              /\//g,
+              '_'
+            )}`,
+            { silent: true }
+          );
+        } catch (_) {
+          // Ignore the error if log file doesn't exist
+        }
+      }
+    }
+    return logsDir;
+  } catch (_) {
+    fs.removeSync(logsDir);
+    return null;
+  }
+}
+
+async function waitForBlockchainNetworkReadiness(mode) {
+  let blockchainSyncAttempt = 1;
+  let blockchainIsReady = false;
+  const maxAttempts = mode === 'single-node' ? 60 : 210;
```

Comment:
Just wanted to check, are these pretty conservative ranges? Any case to make them less/more?

Reply:
Good point @MartinMinkov. I was relying on my Docker Engine allocated resources and on GitHub's CI default hosted runners hardware config.
On my end with 4cpu cores and 8Gb of RAM allocated to Docker Engine it takes < 2 minutes to start network in default mode and ~15 minutes to start network in "closer-to-real" mode.
Similar timing on GH Actions side, it takes < 3 minutes to start default network with HW config like this (Linux):
![image](https://github.com/o1-labs/zkapp-cli/assets/4096154/8d5d261e-68ea-43d8-b1c9-acd0cc2c23b8)

Question though should we implement `maxAttempts calculator` based on the host hardware configuration and resources allocated to Docker Engine (if it is configurable) as part of this PR or it can be done later?

Reply:
>Question though should we implement maxAttempts calculator based on the host hardware configuration and resources allocated to Docker Engine (if it is configurable) as part of this PR or it can be done later?

That would be a great feature but definitely not needed in this PR. Could you create an issue for that, please?

Reply:
https://github.com/o1-labs/zkapp-cli/issues/512

Reply:
+ Agree this will be great but should be done separately.